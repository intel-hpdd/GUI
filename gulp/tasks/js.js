//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
