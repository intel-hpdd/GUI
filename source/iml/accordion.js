// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import { Component } from "inferno";
import { cloneVNode } from "inferno-clone-vnode";

type PanelComponentPropsT = {
  collapsed: boolean,
  accordionId: string,
  panelId: string,
  headingId: string,
  children: React$Element<*>[]
};

type PanelComponentStateT = {
  collapsed: boolean
};

export class PanelComponent extends Component {
  state: PanelComponentStateT;
  props: PanelComponentPropsT;

  constructor(props: PanelComponentPropsT) {
    super(props);

    this.state = {
      collapsed: props.collapsed
    };
  }

  togglePanel() {
    this.setState({
      collapsed: !this.state.collapsed
    });
  }

  render() {
    const title = cloneVNode(this.props.children[0], { collapsed: this.state.collapsed });

    return (
      <div class="panel panel-default">
        <div class="panel-heading" role="tab" id={this.props.headingId}>
          <h4 class="panel-title">
            <a
              className={this.state.collapsed ? "collapsed" : ""}
              role="button"
              data-toggle="collapse"
              data-parent={`#${this.props.accordionId}`}
              aria-expanded={!this.props.collapsed}
              aria-controls={this.props.panelId}
              onClick={this.togglePanel.bind(this)}
            >
              {title}
            </a>
          </h4>
        </div>
        <div
          id={this.props.panelId}
          className={`panel-collapse collapse ${!this.state.collapsed ? "in" : ""}`}
          role="tabpanel"
          aria-labelledby={this.props.headingId}
        >
          <div class="panel-body">{this.props.children[1]}</div>
        </div>
      </div>
    );
  }
}

type AccordionT = {
  id: string,
  children: React$Element<*>
};

export default ({ id, children }: AccordionT) => {
  return (
    <div class="panel-group" id={id} role="tablist" aria-multiselectable="true">
      {children}
    </div>
  );
};
