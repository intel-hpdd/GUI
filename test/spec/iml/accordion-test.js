// @flow

import { default as AccordionComponent, PanelComponent } from "../../../source/iml/accordion.js";
import { render } from "inferno";

describe("accordion", () => {
  let body, div, accordion;
  beforeEach(() => {
    div = document.createElement("div");
    body = document.querySelector("body");

    if (body != null) body.appendChild(div);

    accordion = (
      <AccordionComponent id="test-accordion">
        <PanelComponent collapsed={false} panelId={`collapse0`} headingId={`heading0`} accordionId={"test-accordion"}>
          <h2>Title</h2>
          <>
            <h4>Details:</h4>
            <table class="table">
              <tbody>
                <tr>
                  <td>Created At</td>
                  <td>Jan 01 2019 00:00:00</td>
                </tr>
                <tr>
                  <td>Status</td>
                  <td>Online</td>
                </tr>
              </tbody>
            </table>
          </>
        </PanelComponent>
      </AccordionComponent>
    );

    render(accordion, div);
  });

  afterEach(() => {
    render(null, div);
    if (body != null) body.removeChild(div);
  });

  it("should render", () => {
    expect(div).toMatchSnapshot();
  });
});
