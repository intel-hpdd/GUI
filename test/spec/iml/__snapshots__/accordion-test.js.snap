// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accordion should render 1`] = `
<div>
  <div
    aria-multiselectable="true"
    class="panel-group"
    id="test-accordion"
    role="tablist"
  >
    <div
      class="panel panel-default"
    >
      <div
        class="panel-heading"
        id="heading0"
        role="tab"
      >
        <h4
          class="panel-title"
        >
          <a
            aria-controls="collapse0"
            aria-expanded="true"
            data-parent="#test-accordion"
            data-toggle="collapse"
            role="button"
          >
            <h2
              collapsed="false"
            >
              Title
            </h2>
          </a>
        </h4>
      </div>
      <div
        aria-labelledby="heading0"
        class="panel-collapse collapse in"
        id="collapse0"
        role="tabpanel"
      >
        <div
          class="panel-body"
        >
          <h4>
            Details:
          </h4>
          <table
            class="table"
          >
            <tbody>
              <tr>
                <td>
                  Created At
                </td>
                <td>
                  Jan 01 2019 00:00:00
                </td>
              </tr>
              <tr>
                <td>
                  Status
                </td>
                <td>
                  Online
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
